WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/BaseListUseCase.kt

```diff
@@ -40,15 +42,15 @@ class BaseListUseCase(
             useCases.associateBy { it.type }.mapValues { entry -> entry.value.liveData }
     )
     private val statsTypes = MutableLiveData<List<StatsType>>()
-    val data: MediatorLiveData<UiModel> = mergeNotNull(statsTypes, blockListData) { types, map ->
+    val data: MediatorLiveData<UiModel> = mergeAsyncNotNull(bgDispatcher, statsTypes, blockListData) { types, map ->
         types.mapNotNull {
             if (map.containsKey(it)) {
                 map[it]
             } else {
                 null
             }
         }
-    }.map { useCaseModels ->
+    }.mapAsync(bgDispatcher) { useCaseModels ->
```

Comment:
👍 I wonder why we didn't think of making this async sooner 😃.

Reply:
for me it was the "this is like RxJava" approach where I considered everything in the stream to run on a background thread :-) - an expensive mistake